# BioE131FinalProject
step 1: setup you AWS acount. please refer to the instructions listed on Lab 8 on how to setup your AWS acount.
step 2: setup Jbrowse according to the isntructions listed on Lab 8. 
please verig that JBrowse is installed using the following bash command: --version Jbrowse 
run jbrowse --help to check the following packages and commands are installed
wget: For downloading genome and annotation files.
samtools: For indexing genome files.
minimap2: For sequence alignment. // ... check qith versions ...
JBrowse CLI: For adding assemblies and tracks.

step 3 : Ensure AWS enviroment is set up by running the following command : export APACHE_ROOT=/var/www/html
step 4: make sure ubuntu has the correct permissions : sudo chown -R $(whoami) $APACHE_ROOT/jbrowse2
requried tools : 
verify the following tools are installed:

wget: For downloading genome and annotation files.
samtools: For indexing genome files.
minimap2: For sequence alignment.
JBrowse CLI: For adding assemblies and tracks.

by running the following commands: 
wget --version
samtools --version
minimap2 --version
jbrowse --version

(please note viral_genome is a place holder and you must plug in your file name for the following steps)
step 5: using the wget -O command we will download the gene
step 6 : then we wil ldecompres the .gz file using "gunzip viral_genome.fna.gz" 
step 7 : index the genome file using samtools with the following bash command: samtools faidx viral_genome.fna
step 8 : add the file to the Jbrowse : jbrowse add-assembly viral_genome.fna --out $APACHE_ROOT/jbrowse2 --load copy
to add anotations for the genes use the following commands: 
step 9 : wget -O annotations.bed.gz <put link here>
step 10: decompress and prepare the BED file
gunzip filoviridae_annotations.bed.gz
jbrowse sort-bed filoviridae_annotations.bed > sorted.bed
bgzip sorted.bed
tabix sorted.bed.gz
step 11: add the BED anotaion to JBrowse:
jbrowse add-track sorted.bed.gz --out $APACHE_ROOT/jbrowse2 --load copy --assemblyNames KM034562v1


to align two genome sequences using minimap2 follow the following instructions : 


step 12: using minimap2 we will allign the sequences
minimap2 -x asm5 -c reference_genome.fasta query_genome.fasta > output.paf

use the following if a SAM format is required : minimap2 -a reference_genome.fasta query_genome.fasta > output.sam

after setting up follow the following commands to make sure everything is running smoothly 



follow the following provided bash commands to get setup and to get our genes and anotations ready using the provided explanation:

2000EboVirSequence: 
```bash
wget -O 2000EboVirSequence.fna.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/848/505/GCF_000848505.1_ViralProj14703/GCF_000848505.1_ViralProj14703_genomic.fna.gz 
gunzip 2000EboVirSequence.fna.gz
samtools faidx 2000EboVirSequence.fna
jbrowse add-assembly 2000EboVirSequence.fna --out $APACHE_ROOT/jbrowse2 --load copy
```


2000EboVirSequence anotations: 
```bash

wget -O 2000EboVirSequence_annotations.gff3.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/848/505/GCF_000848505.1_ViralProj14703/GCF_000848505.1_ViralProj14703_genomic.gff.gz 
gunzip 2000EboVirSequence_annotations.gff3.gz
jbrowse sort-gff 2000EboVirSequence_annotations.gff3 > genes.gff
bgzip genes.gff
tabix genes.gff.gz
jbrowse add-track genes.gff.gz --out $APACHE_ROOT/jbrowse2 --load copy
Add â€”force
jbrowse text-index --out $APACHE_ROOT/jbrowse2
```



2014EboVirSequence : 
```bash

wget -O 2014EboVirSequence.fna.gz ftp://hgdownload.soe.ucsc.edu/goldenPath/eboVir3/bigZips/KM034562v1.fa.gz
gunzip 2014EboVirSequence.fna.gz
samtools faidx 2014EboVirSequence.fna
```



annotations for  2014EboVirSequence : 
```bash

wget -O 2014EboVirSequence_annotations.bed usegalaxy.org.au/api/datasets/a6e389a98c2d16789414434a46f1f1ba/display?to_ext=bed
jbrowse sort-bed 2014EboVirSequence_annotations.bed > sorted.bed
bgzip sorted.bed
tabix sorted.bed.gz
jbrowse add-track sorted.bed.gz --out $APACHE_ROOT/jbrowse2 --load copy --assemblyNames 2014.fna
jbrowse text-index --out $APACHE_ROOT/jbrowse2 
jbrowse text-index --assemblyNames 2014EboVirSequence_genome --out indexed.bed --file sorted2.bed
```





to add ICTV figure 4A:
```bash

wget -O ICTV4A_genome.fna https://ftp.ictv.global/sites/default/files/report_files/Filo_Fig4.v3.CG_alignment.TXT --no-check-certificate
https://www.hiv.lanl.gov/cgi-bin/FORMAT_CONVERSION/convert.cgi.gz
ftp://ictv.global/sites/default/files/report_files/Filo_Fig4.v3.CG_alignment.TXT
Format converter: Download (https://www.hiv.lanl.gov/cgi-bin/FORMAT_CONVERSION/convert.cgi)
```



to add the GenBank : ASM3409842v1 ( 2015EboVirSequence)
```bash

wget -O  2015EboVirSequence.fna.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/034/098/425/GCA_034098425.1_ASM3409842v1/GCA_034098425.1_ASM3409842v1_genomic.fna.gz
gunzip  2015EboVirSequence.fna.gz
samtools faidx  2015EboVirSequence.fna
jbrowse add-assembly   2015EboVirSequence.fna --out $APACHE_ROOT/jbrowse2 --load copy
```
adding anotations for : ASM3409842v1 ( 2015EboVirSequence)
```bash
wget -O  2015EboVirSequence_annotations.gff3.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/034/098/425/GCA_034098425.1_ASM3409842v1/GCA_034098425.1_ASM3409842v1_genomic.gff.gz 
gunzip  2015EboVirSequence_annotations.gff3.gz
jbrowse sort-gff  2015EboVirSequence_annotations.gff3 >  2015EboVirSequence_sorted.gff
bgzip  2015EboVirSequence_sorted.gff
tabix  2015EboVirSequence_sorted.gff.gz
jbrowse add-track  2015EboVirSequence_sorted.gff.gz --out $APACHE_ROOT/jbrowse2 --load copy --assemblyNames  2015EboVirSequence_anotated.fna
jbrowse text-index --out $APACHE_ROOT/jbrowse2
```

adding GenBank: ASM90009415v1 (2016EboVirSequence)
```bash

wget -O 2016EboVirSequence.fna.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/094/155/GCA_900094155.1_ASM90009415v1/GCA_900094155.1_ASM90009415v1_genomic.fna.gz 
gunzip 2016EboVirSequence.fna.gz 
samtools faidx 2016EboVirSequence.fna
jbrowse add-assembly  2016EboVirSequence.fna --out $APACHE_ROOT/jbrowse2 --load copy
```
Adding annotations for ASM90009415v1
```bash

wget -O 2016EboVirSequence_annotations.gff3.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900/094/155/GCA_900094155.1_ASM90009415v1/GCA_900094155.1_ASM90009415v1_genomic.gff.gz
gunzip 2016EboVirSequence_annotations.gff3.gz
jbrowse sort-gff 2016EboVirSequence_annotations.gff3 > 2016EboVirSequence_sorted.gff
bgzip 2016EboVirSequence_sorted.gff
tabix 2016EboVirSequence_sorted.gff.gz
jbrowse add-track 2016EboVirSequence_sorted.gff.gz --out $APACHE_ROOT/jbrowse2 --load copy --assemblyNames 2016EboVirSequenceAnotated.fna
jbrowse text-index --out $APACHE_ROOT/jbrowse2
```



























